System/Primitive.cpp

```diff
@@ -158,6 +158,102 @@ namespace RTE {
 		}
 	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void RoundBoxPrimitive::Draw(BITMAP *drawScreen, Vector targetPos) {
+		if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) {
+			Vector drawStart = m_StartPos - targetPos;
+			Vector drawEnd = m_EndPos - targetPos;
+
+			// Top left corner
```

Comment:
You can probably combine these 4 comments into 1. Same thing with the plane comments below, i.e.
```
// Draw the top left, bottom left, top right and bottom right corners respectively
arc(...);
arc(...);
arc(...);
arc(...);

//Draw the top, bottom, left and right planes respectively
hline(...);
hline(...);
vline(...);
vline(...);
```
Alternatively, you could combine them all into one line, or you can get rid of them and just make people spend a few seconds reading, up to you.